core/model/src/main/java/org/eclipse/rdf4j/model/impl/SimpleLiteral.java

```diff
@@ -138,12 +158,16 @@ protected void setLanguage(String language) {
 			throw new IllegalArgumentException("Language tag cannot be empty");
 		}
 		this.language = language;
+		this.optionalLanguage = Optional.of(language);
 		setDatatype(RDF.LANGSTRING);
 	}
 
 	@Override
 	public Optional<String> getLanguage() {
-		return Optional.ofNullable(language);
+		if (optionalLanguage == null) {
+			optionalLanguage = Optional.ofNullable(language);
+		}
+		return optionalLanguage;
 	}
 
 	protected void setDatatype(IRI datatype) {
```

Comment:
Having protected setters for everything is a real pain in the backside. It means we can't assume that the object is immutable which is why I've had to implement a "lateInit" boolean.